src/app/_components/SearchBar.tsx

```diff
@@ -0,0 +1,55 @@
+"use client";
+
+import React from "react";
+import FiltersIcon from "./icons/filters";
+import SearchIcon from "./icons/search";
+import { usePathname, useRouter, useSearchParams } from "next/navigation";
+
+const SearchBar: React.FC = () => {
+  const searchParams = useSearchParams();
+  const router = useRouter();
+  const pathname = usePathname();
+
+  const handleTextSearch = (text: string) => {
+    const params = new URLSearchParams(searchParams);
+    if (text) {
+      params.set("query", text);
+    } else {
+      params.delete("query");
+    }
```

Comment:
We can a blank line before the if, and also we can remove the brackets because they're not necessary
```suggestion
    const params = new URLSearchParams(searchParams);
    
    if (text) params.set("query", text);
    else params.delete("query");
```